code/further_analyses/ica_analysis.py

```diff
@@ -0,0 +1,103 @@
+"""
+A set of functions for extracting and visualizing results from applying independent components analysis (ICA) on the
+preprocessed neuroimaging data.
+"""
+
+import os
+import numpy as np
+import nibabel as nib
+import matplotlib.pyplot as plt
+
+def prep_data(data_path='/Users/nimahejazi/Dropbox/159_projData/preprocessed/',subj_num):
+  BOLD_file_1 = os.path.join(data_path,'sub'+subj_num+'_BOLD','task001_run001/filtered_func_data_mni'+'.'+'nii.gz')
+  BOLD_file_2 = os.path.join(data_path,'sub'+subj_num+'_BOLD','task002_run001/filtered_func_data_mni'+'.'+'nii.gz')
+  BOLD_file_3 = os.path.join(data_path,'sub'+subj_num+'_BOLD','task003_run001/filtered_func_data_mni'+'.'+'nii.gz')
+
+  img_1 = nib.load(BOLD_file_1)
+  data_1 = img_1.get_data()
+  data_1 = data_1[..., 5:]
+
+  img_2 = nib.load(BOLD_file_2)
+  data_2 = img_2.get_data()
+  data_2 = data_2[..., 5:]
+
+  img_3 = nib.load(BOLD_file_3)
+  data_3 = img_3.get_data()
+  data_3 = data_3[..., 5:]
+
+### Preprocess ################################################################
+from nilearn.input_data import NiftiMasker
+
+# Remove the background with mask_strategy='epi' to compute the mask from the EPI images
+
+def clean_data():
+  masker = NiftiMasker(smoothing_fwhm=8, memory='nilearn_cache', memory_level=1,
```

Comment:
there is a prepare_mask function in the new general_utils.py (again, need to merge my pull request to see this). That function returns in_brain_mask. 

It is exactly the same as in this lecture material: http://www.jarrodmillman.com/rcsds/lectures/modeling_noise.html  If you read it, you can make use the residuals from pca. 

I removed the first two PCs because they represent more anatomical features of the brain and noise. If they are not removed, we might not be able to get good independent components. Could you look into the mask and ask or call me if you have questions? Thanks!